src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxResolvingClassLoader.java

```diff
@@ -63,38 +61,34 @@
 
     @Override public URL getResource(String name) {
         ClassLoader parentLoader = getParent();
-        return load(parentResourceCache, name, parentLoader, () -> Optional.fromNullable(parentLoader.getResource(name))).orNull();
+        return load(parentResourceCache, name, parentLoader, item -> Optional.fromNullable(parentLoader.getResource(item))).orNull();
     }
 
     // We cannot have the inner cache be a LoadingCache and just use .get(name), since then the values of the outer cache would strongly refer to the keys.
-    private static <T> T load(LoadingCache<ClassLoader, Cache<String, T>> cache, String name, ClassLoader parentLoader, Supplier<T> supplier) {
-        try {
-            return cache.getUnchecked(parentLoader).get(name, () -> {
-                Thread t = Thread.currentThread();
-                String origName = t.getName();
-                t.setName(origName + " loading " + name);
-                long start = System.nanoTime(); // http://stackoverflow.com/q/19052316/12916
-                try {
-                    return supplier.get();
-                } finally {
-                    t.setName(origName);
-                    long ms = (System.nanoTime() - start) / 1000000;
-                    if (ms > 1000) {
-                        LOGGER.log(Level.INFO, "took {0}ms to load/not load {1} from {2}", new Object[] {ms, name, parentLoader});
-                    }
+    private static <T> T load(LoadingCache<ClassLoader, Cache<String, T>> cache, String name, ClassLoader parentLoader, Function<String, T> mapping) {
+        return cache.get(parentLoader).get(name, (String item) -> {
```

Comment:
Caffeine did not want to accept a Supplier.

Reply:
I guess `item == name` here?